### PR TITLE
Fix: fix for #18257 and #17391

### LINKED ISSUE #17391: Relation filed search not working as before + wrong layout
### Required System information

- Node.js version: v16.20.1
- NPM version: 7.24.2
- Strapi version: 4.11.7
- Database: postgres
- Operating system: macOS Ventura
- Is your project Javascript or Typescript: ts

### Describe the bug

When searching in relation field, it returns results matching only beginning of the "relation display field". Before, it matched all results.

E.g. we have childPages, using `path` as main relations display field. Typing `ubytovanie` should return `bratislava-pre-ukrajinu/ubytovanie` but it doesn't.

Choosing some relations manually works. Search matching the beginning works.

But the layout is somehow broken.

### Expected behavior

Search in relations field should work correctly. The layout should display correctly.
It would be nice if it also ignores accents (áéšťýö etc...)

### Screenshots

image
image

image

### LINKED ISSUE #18257: search in relation fields does not work properly
### Required System information
- Node.js version: 18.14.2
- NPM version: 9.5.0
- Strapi version:4.12.5
- Database:postgres
- Operating system: windows
- Is your project Javascript or Typescript: js

### Describe the bug

Screenshot 2023-10-03 151056
Screenshot 2023-10-03 151106
Screenshot 2023-10-03 151115
Screenshot 2023-10-03 151127

### Expected behavior
it would be nice to see data when each letter is typed (or maybe at least 3 letter)

### PR DESCRIPTION
**Pull request by Arshiash80**
### What does it do?

This PR addresses and fixes two specific issues: #17391 and #18257. The main focus is on resolving the problem related to searching within a relational input.

### Why is it needed?

The issue #17391 and #18257 highlighted a problem with searching in a relational input. This PR aims to solve this issue by making necessary changes.

### How to test it?

To verify the behavior of this fix, follow these steps:

1. Set up the environment with the changes from this PR.
2. Navigate to the relational input that was problematic (context of the mentioned issues).
3. Perform a search operation and observe the behavior.
4. Confirm that the autocomplete on the combobox now uses "none" and the filter logic is changed to "_filter: '$containsi'".

### Related issue(s)/PR(s)

This PR is related to the following issues:
- Fixes #17391
- Fixes #18257

pr: ready

### CODE REVIEW

**Comment by in-vbonilla** [comment]
We encountered this error today and found the PR fresh from the oven! 

Thanks!

**Comment by Arshiash80** [comment]
@in-vbonilla Thanks for the quick review! I'm glad you found the PR fresh from the oven :)

**Comment by joshuaellis** [comment]
Thanks for the contribution! There's a few things to do before this PR is ready:

* It needs to be based off the `develop` branch and aimed at that branch too. You'll also find the affected files are in typescript now.
* The title should say what it does, not just "fix for issue XXXX" because that's not very helpful in quick reading, I would suggest describing what you've done
* The "What does it do?" needs to be done correctly as well please, it should list at a high level the changes you've done
* You also have failing tests, these need to be fixed as well please

I've tested the change and it sort of works, I have a weird behaviour and it's probably easier to view the recording below: 

https://github.com/strapi/strapi/assets/37798644/4a534767-2640-47ca-ba09-c60efb7b8b36

the TLDR; is the filter list is not being updated correctly. If we can sort all these things out then this PR will be ready 😄

**Comment by joshuaellis** [comment]
closing in favour of #19043
